src/RegionHintRecommender.hpp

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2015 - 2023, Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#ifndef REGIONHINTRECOMMENDER_HPP_INCLUDE
+#define REGIONHINTRECOMMENDER_HPP_INCLUDE
+
+#include <memory>
+#include <map>
+
+namespace geopm
+{
+    /// @brief Class ingesting the output from a DomainNetMap and
+    ///        a frequency map json file and determining a recommended 
+    ///        frequency decision.
+    class RegionHintRecommender
+    {
+        public:
+            /// @brief Returns a unique_ptr to a concrete object constructed
+            ///        using the underlying implementation, which loads a
+            ///        frequency map file into a std::map of region class
+            ///        string to double, and sets both min and max frequency
+            ///        recommendations.
+            ///
+            /// @param [in] fmap_path string containing path to freq map json
+            /// @param [in] min_freq integer containing minimum frequency
+            /// @param [in] max_freq integer containing maximum frequency
+            static std::unique_ptr<RegionHintRecommender> make_unique(const std::string &fmap_path,
+                                                                      int min_freq, int max_freq);
+            /// @brief Returns a shared_ptr to a concrete object constructed
+            ///        using the underlying implementation, which loads a
+            ///        frequency map file into a std::map of region class
+            ///        string to double, and sets both min and max frequency
+            ///        recommendations.
+            ///
+            /// @param [in] fmap_path string containing path to freq map json
+            /// @param [in] min_freq integer containing minimum frequency
+            /// @param [in] max_freq integer containing maximum frequency
+            static std::shared_ptr<RegionHintRecommender> make_shared(const std::string &fmap_path,
+                                                                      int min_freq, int max_freq);
+
+            virtual ~RegionHintRecommender() = default;
+            /// @brief Recommends frequency based on the output from a DomainNetMap neural net
+            ///        evaluation
+            ///
+            /// @param [in] nn_output Output from the DomainNetMap neural net evaluation, as a
+            ///             map from trace names to doubles.
+            /// @param [in] phi User-input perf-energy bias
```

Comment:
Need a docstring here explaining the output.  It's in Hz, which might be unexpected.

Reply:
I don't think the output here is in Hz? It's a distribution of probabilities. Should be unitless. I was trying to keep this more generic and have specific implementation details in RegionHintRecommendorImp.hpp